src/test/ui/borrowck/issue-45983.stderr

```diff
@@ -0,0 +1,12 @@
+error: borrowed data cannot be moved outside of its closure
+  --> $DIR/issue-45983.rs:17:27
+   |
+16 |     let x = None;
+   |         - borrowed data cannot be moved into here...
+17 |     give_any(|y| x = Some(y));
+   |              ---          ^ cannot be moved outside of its closure
+   |              |
+   |              ...because it cannot outlive this closure
+
+error: aborting due to previous error
```

Comment:
can you add a `ui` test for the other case you mentioned? The one where you push into a vector that exists outside the closure?

Reply:
Moved existing test that shows the behavior to ui: https://github.com/rust-lang/rust/pull/47144/files#diff-e5107c6d2a404ecd34f782e9e9e8b20f